### PR TITLE
Adds docker deployment of glados

### PR DESCRIPTION
**Pull request by mrferris**
Creates a deployment containing:
- Glados Audit
- Glados Monitor
- Glados Web
- Trin
- Postgres

- [x] Add docs
- [x] Push docker images to dockerhub

### CODE REVIEW

File: DOCKER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Docker Guide
+
+The `docker-compose.yml` file defines a glados deployment containing a trin node,
+a postgres instance, and the components of glados pre-configured to use the node and postgres
+instance.
+
+### Deploying Glados
+
+First, set the environment variable `GLADOS_POSTGRES_DATA_DIR` to the directory
+where Postgres should store its data.
```
Comment (carver) [inline DOCKER_GUIDE.md:10]: It's probably pretty obvious, but nice to be explicit.

```suggestion
where Postgres should store its data on the host machine.
```

**Review by carver — APPROVED** [review]
I'm no docker expert, but I didn't notice anything problematic. 👍🏻 

This will be nice for both local runs and getting glados into new infrastructure!

File: glados-web/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+RUN apt-get install clang -y
+
+RUN apt-get install -y --no-install-recommends ca-certificates
+RUN update-ca-certificates
+
+COPY . .
+
+RUN cargo build -p glados-web --release
+
+FROM ubuntu:20.04
```
Comment (KolbyML) [inline glados-web/Dockerfile:21]: Is there a reason to use 20?

File: glados-monitor/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+RUN apt-get install clang -y
+
+RUN apt-get install -y --no-install-recommends ca-certificates
+RUN update-ca-certificates
+
+COPY . .
+
+RUN cargo build -p glados-monitor --release
+
+FROM ubuntu:20.04
```
Comment (KolbyML) [inline glados-monitor/Dockerfile:21]: Same here 20 question?

File: glados-audit/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+RUN apt-get install clang -y
+
+RUN apt-get install -y --no-install-recommends ca-certificates
+RUN update-ca-certificates
+
+COPY . .
+
+RUN cargo build -p glados-audit --release
+
+FROM ubuntu:20.04
```
Comment (KolbyML) [inline glados-audit/Dockerfile:21]: Same here 20 question?

**Comment by pauljickling** [comment]
> I'm no docker expert, but I didn't notice anything problematic. 👍🏻
> 
> This will be nice for both local runs and getting glados into new infrastructure!

Agreed. This lgtm!

**Comment by mrferris** [comment]
Hitting an SSL error in the `glados-monitor` container, which is unexpected. Fixing that then merging.

**Pull request merged**